metricshub-agent/src/main/java/org/metricshub/web/config/ToolCallbackConfiguration.java

```diff
@@ -80,7 +83,8 @@ public ToolCallbackProvider metricshubTools(
 				hostDetailsService,

```

Comment:
Can we change this implementation to 

```java
@Bean
public ToolCallbackProvider metricshubTools(ApplicationContext context) {
    // Automatically get all beans of type ToolInterface
    Map<String, ToolInterface> tools = context.getBeansOfType(IToolInterface.class);

    return MethodToolCallbackProvider.builder()
            .toolObjects(tools.values())
            .build();
}
```

And make all the tools implementing IToolInterface.